packages/server/src/restaurants/database/DishesSchema.ts

```diff
@@ -0,0 +1,185 @@
+import { Prop, raw, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Exclude, Expose } from 'class-transformer';
+import { ObjectId } from 'mongodb';
+import mongoose, { Document } from 'mongoose';
+
+type DishDocument = Dish & Document<ObjectId>;
+
+enum DishTags {
+  Vegan = 'wegańska',
+  Vegetarian = 'wegetariańska',
+  GlutenFree = 'gluten free',
+  Spicy = 'ostre',
+  VerySpicy = 'bardzo ostre',
+}
+
+enum MealType {
+  Breakfast = 'śniadanie',
+  Lunch = 'lunch',
+  Dinner = 'obiad',
+  Supper = 'kolacja',
+}
+
+enum Allergens {
+  Gluten = 'gluten',
+  ShellFish = 'skorupiaki',
+  Eggs = 'jaja',
+  Fish = 'ryby',
+  Nuts = 'orzechy',
+  Soya = 'soja',
+  Milk = 'mleko',
+  Celery = 'seler',
+  Mustard = 'gorczyca',
+  SesameSeeds = 'nasiona sezamu',
+  SulphurDioxide = 'dwutlenek siarki',
+  Lupine = 'łubin',
+  Molluscs = 'mięczaki',
+}
+
+const DISH_CONSTANTS = Object.freeze({
+  DISH_LEN_STR: Object.freeze({
+    MIN_LENGTH: 1,
+    MAX_LENGTH: 500,
+  }),
+  DISH_LEN_NUMBER: Object.freeze({
+    MIN_LENGTH: 1,
+    MAX_LENGTH: 5,
+  }),
+});
+
+@Exclude()
+@Schema({
+  collection: 'dishes',
+})
+class Dish {
+  @Expose()
+  @Prop({
+    required: true,
+    minlength: DISH_CONSTANTS.DISH_LEN_STR.MIN_LENGTH,
+    maxlength: DISH_CONSTANTS.DISH_LEN_STR.MAX_LENGTH,
+  })
+  name: string;
+
+  @Expose()
+  @Prop({
+    type: [{ enum: MealType }],
+    minlength: DISH_CONSTANTS.DISH_LEN_STR.MIN_LENGTH,
+    maxlength: DISH_CONSTANTS.DISH_LEN_STR.MAX_LENGTH,
+  })
+  mealType: string[];
+
+  @Expose()
+  @Prop({
+    minlength: DISH_CONSTANTS.DISH_LEN_STR.MIN_LENGTH,
+    maxlength: DISH_CONSTANTS.DISH_LEN_STR.MAX_LENGTH,
+  })
+  description: string;
+
+  @Expose()
+  @Prop({
+    required: true,
+    minlength: DISH_CONSTANTS.DISH_LEN_NUMBER.MIN_LENGTH,
+    maxlength: DISH_CONSTANTS.DISH_LEN_NUMBER.MAX_LENGTH,
+  })
+  price: mongoose.Decimal128;
```

Comment:
Co do ceny to ustaliliśmy, że będziemy trzymać ją w groszach (czyli wartość 2350 odpowiada 23,50PLN). Więc tu musiałoby być ograniczenie, że musi to być liczba całkowita (no i może jeszcze jakaś wartość max)